docs/monitoring/autodiscovery.md

```diff
@@ -0,0 +1,60 @@
+
+### How does Sematext Autodiscovery Monitoring work in bare-metal/virtualMachine environments?
+
+It is enough to install sematext-agent rpm/deb package and set up one of your Infra App tokens (just follow UI instructions for this). This will start Sematext Agent which will
+automatically start scanning for known service types. Separately, it will connect to Sematext backend to fetch info about any automatic monitoring rules you may have defined in UI.
+
+In bare-metal/vritualMachine environments, Sematext Agent will automatically create monitoring agents only if configured so in the UI.
+
+### How does Sematext Autodiscovery Monitoring work in container environments? 
+
+The Sematext Agent container gets distributed to every Kubernetes, Docker Swarm or Docker Enterprise node. Then it starts watching container events and compares the image or process name of started containers with a list of application names defined in a configuration template. Then Sematext Agent connects the container network and configures Sematext Agent for the right IP-address and port and other settings according to the configuration template.
+
+In container environments, Sematext Agent will create a monitoring agent if it finds MONITORING_TOKEN in environment variables or labels of your service or if you enabled automatic monitoring for specific service type in the UI. 
+
+Example: Steps required to monitor a containerized application like Elasticsearch manually: 
+1. Visit SematextCloud UI, create new Elasticsearch App, open container env instructions
+2. Following the UI instructions, deploy Sematext Agent container (as a standalone container, as Kubernetes DaemonSet or as global Swarm Service)
+3. Add the Elasticsearch MONITORING_TOKEN as label or environment variable to your Elasticsearch container
+4. Restart your Elasticsearch container to apply the label/environment variable
+
+Example: Steps required to monitor a containerized application like Elasticsearch automatically: 
+1. Visit Discovery in UI - open instructions to install Sematext Agent in container env
+2. Following the UI instructions, deploy Sematext Agent container (as a standalone container, as Kubernetes DaemonSet or as global Swarm Service)
+3. From Discovery UI, click on any of discovered services and enable automatic monitoring option
+
+As soon Sematext Agent discovers the Elasticsearch container (having a MONITORING_TOKEN set or having automatic monitoring option enabled) it will set up monitoring agent container which will start to collect the Elasticsearch metrics. 
```

Comment:
As soon Sematext Agent discovers the Elasticsearch container (having a MONITORING_TOKEN set or having automatic monitoring option enabled) -> As soon Sematext Agent discovers the Elasticsearch container having a `MONITORING_TOKEN` set or automatic monitoring enabled